core/src/main/java/org/apache/accumulo/core/spi/scan/ScanInfo.java

```diff
@@ -125,4 +125,9 @@ enum Type {
    * @return Hints set by a scanner using {@link ScannerBase#setExecutionHints(Map)}
    */
   Map<String,String> getExecutionHints();
+
+  /**
+   * @return user data set by a scanner using {@link ScannerBase#setUserData(String)}
+   */
+  String getUserData();
```

Comment:
Since this is adding a method to an API/SPI interface, it would be nice to have a default method here, so it doesn't immediately break all users who implemented this SPI in the previous release.

Reply:
default implementation returns an empty string now